docs/content/docs/release_guide.md

```diff
@@ -37,55 +39,45 @@ Push the release branch to the main repo (not fork), identified here by the `ups
+* The default osm-controller image tag in [osm cli](https://github.com/openservicemesh/osm/blob/main/cmd/cli/install.go) to point to the new release
+* The Helm chart [README.md](https://github.com/openservicemesh/osm/blob/main/charts/osm/README.md)
+  - Necessary changes should be made automatically by running `make chart-readme`
+
+Once patches and version information have been updated on a new branch off of the release branch, create a pull request from the new branch to the release branch. Proceed to the next step once the pull request is approved and merged.
 
 ## Create and push a Git tag
 
-Once the release is ready to be published, create and push a Git tag from the release branch in
+Ensure your local copy of the release branch has the latest changes from the PR merged above.
+
+Once the release is ready to be published, create and push a Git tag from the release branch to
 the main repo (not fork), identified here by the `upstream` remote.
 
 ```console
 $ export RELEASE_VERSION=<release-version> # ex: export RELEASE_VERSION=v0.4.0
-$ git tag -a "$RELEASE_VERSION" -m "<add description here>"
+$ git tag "$RELEASE_VERSION"
 $ git push upstream "$RELEASE_VERSION"
 ```
 
 A [GitHub Action](https://github.com/openservicemesh/osm/blob/main/.github/workflows/release.yml) is triggered when the tag is pushed.
 It will build the CLI binaries, publish a new GitHub release,
-upload the packaged binaries and checksums as release assets,
-and build and push Docker images for OSM and the demo to the
-[`openservicemesh` organization](https://hub.docker.com/u/openservicemesh) on Docker Hub.
+upload the packaged binaries and checksums as release assets, build and push Docker images for OSM and the demo to the
+[`openservicemesh` organization](https://hub.docker.com/u/openservicemesh) on Docker Hub, and publish the Helm chart to the repo hosted at https://openservicemesh.github.io/osm.
 
 ## Add release notes
 
 In the description section of the new release, add information about feature additions, bug fixes,
 and any other administrative tasks completed on the repository.
 
-## Update README chart with helm-docs
-
-Chart's README is autogenerated with [helm-docs](https://github.com/norwoodj/helm-docs). Once versioning has been addressed
-under `charts/osm/Chart.yaml`, update the README with the following command:
-
-```
-make chart-readme
-```
-
-## Create a Helm chart release
-
-* create a Git tag from main in the format `chart/x.x.x` where `x.x.x` is the chart version
-* push the Git tag to the main repo (not fork)
-* GitHub Actions runs [helm-gh-pages](https://github.com/stefanprodan/helm-gh-pages) that packages the charts, updates `index.yaml` and pushes the changes to `gh-pages` branch
-* GitHub Pages publishes the new version, making the chart available at `https://openservicemesh.github.io/osm`
-
 ## Announce the new release
 
 Make an announcement on the [mailing list](https://groups.google.com/g/openservicemesh).
```

Comment:
Can we add to post an announcement in Slack or is the audience pretty much the same?

Reply:
I think the audience is likely very much the same, but since we've been announcing on Slack recently too, I'll add that.